cargo-miri/bin.rs

```diff
@@ -888,7 +936,14 @@ fn phase_rustc(mut args: impl Iterator<Item = String>, phase: RustcPhase) {
                 cmd.arg("--emit=metadata");
             }
 
-            cmd.args(&env.args);
+            let mut args = env.args.clone();
```

Comment:
```suggestion
            // Alter the `-o` parameter so that it does not overwrite the JSON file we stored above.
            let mut args = env.args.clone();
```